File: backend/app/schemas/file.py
"""File response schemas."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class FileUploadResponse(BaseModel):
    """Response schema for file upload."""

    id: UUID
    original_filename: str
    file_size: int
    file_type: str
    created_at: datetime
    data_summary: str | None = None
    user_context: str | None = None

    model_config = ConfigDict(from_attributes=True)


class FileListItem(BaseModel):
    """Response schema for file list item."""

    id: UUID
    original_filename: str
    file_size: int
    file_type: str
    created_at: datetime
    updated_at: datetime
    has_summary: bool = False

    model_config = ConfigDict(from_attributes=True)


class FileDetailResponse(BaseModel):
    """Response schema for file detail."""

    id: UUID
    original_filename: str
    stored_filename: str
    file_size: int
    file_type: str
    created_at: datetime
    updated_at: datetime
    data_summary: str | None = None
    user_context: str | None = None

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/file.ts
/**
 * File request and response types.
 * These types mirror backend schemas in backend/app/schemas/file.py
 */

export interface FileUploadResponse {
  id: string;
  original_filename: string;
  file_size: number;
  file_type: string;
  created_at: string;
  data_summary: string | null;
  user_context: string | null;
}

export interface FileListItem {
  id: string;
  original_filename: string;
  file_size: number;
  file_type: string;
  created_at: string;
  updated_at: string;
  has_summary: boolean;
}

export interface FileDetailResponse {
  id: string;
  original_filename: string;
  stored_filename: string;
  file_size: number;
  file_type: string;
  created_at: string;
  updated_at: string;
  data_summary: string | null;
  user_context: string | null;
}

export interface FileSummaryResponse {
  data_summary: string | null;
  user_context: string | null;
  query_suggestions: {
    categories: Array<{
      name: string;
      queries: string[];
    }>;
  } | null;
  suggestion_auto_send: boolean;
}
